### PR TITLE
Refactor Artifacts at rest for safe/easy deployment

### PR DESCRIPTION
**Pull request by scbedd**
No more of this `stage-filtered-artifacts` stuff. We'll generate the whl/sdist/documents into the appropriate folder within the artifact, categorized by package name like we see over in .NET.

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-python-release.yml [hunk 1]
```diff
@@ -40,24 +41,26 @@ stages:
...
                           ForRelease: true
-                    - template: /eng/pipelines/templates/steps/stage-filtered-artifacts.yml
-                      parameters:
-                        SourceFolder: ${{parameters.ArtifactName}}
-                        TargetFolder: ${{artifact.safeName}}
-                        PackageName: ${{artifact.name}}
+
                     - pwsh: |
-                        Get-ChildItem -Recurse $(Pipeline.Workspace)/${{artifact.safeName}}
+                        $packageDirectory = "${{artifact.name}}".Replace("_", "-")
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-python-release.yml:53]: Do we need to do this replacement? Can we just use the artifact name?

Comment (scbedd) [reply eng/pipelines/templates/stages/archetype-python-release.yml:53]: Discussed offline. This is placed in lieu of touching every `ci.yml` in this PR. Will work with an admin to push the updating commit directly to master after this PR goes in.

**Review by weshaggard — APPROVED** [review]
Changes look reasonable based on our offline discussion about the artifact names being switched to match the package name in a follow-up.

File: eng/pipelines/templates/stages/archetype-python-release.yml [hunk 2]
```diff
@@ -280,22 +283,17 @@ stages:
...
+              echo "##vso[task.setvariable variable=Package.Name]$packageDirectory"
+          - pwsh: |
+              $fileCount = (Get-ChildItem $(Pipeline.Workspace)/${{parameters.ArtifactName}}/$packageDirectory  | Measure-Object).Count
               if ($fileCount -eq 0) {
                 Write-Host "No alpha packages for ${{artifact.safeName}} to publish."
                 exit 0
               }
 
-              twine upload --repository $(DevFeedName) --config-file $(PYPIRC_PATH) $(Pipeline.Workspace)/${{artifact.safeName}}/*a*.whl
+              twine upload --repository $(DevFeedName) --config-file $(PYPIRC_PATH) $(Pipeline.Workspace)/${{parameters.ArtifactName}}/$(Package.Name)/*a*.whl
```
Comment (mitchdenny) [inline eng/pipelines/templates/stages/archetype-python-release.yml:295]: Not really what this change is about, but I wonder if this wildcard is too permissive.

**Review by mitchdenny — APPROVED** [review]
Looks reasonable. I think it might be worth merging this onto a feature branch in upstream and testing the release process end to end on template.

File: eng/pipelines/templates/stages/archetype-python-release.yml (hunk 2 above)
Comment (scbedd) [reply eng/pipelines/templates/stages/archetype-python-release.yml:295]: What makes you say that?

**Comment by scbedd** [comment]
@mitchdenny already done here: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=747165&view=logs&j=3c922a7d-1305-5297-ba3f-884e0ad84366

**Pull request merged**